activiti-core/activiti-engine/src/main/java/org/activiti/engine/impl/cmd/DeployCmd.java

```diff
@@ -124,9 +127,9 @@ private void setProjectReleaseVersion(DeploymentEntity deployment) {
     protected boolean deploymentsDiffer(DeploymentEntity deployment,
                                         DeploymentEntity saved) {
 
-        if (deploymentBuilder.hasProjectManifestSet()) {
+        if (deploymentBuilder.hasEnforcedAppVersion()) {
+            return !deploymentBuilder.getEnforcedAppVersion().equals(saved.getVersion());
 
-            return !deployment.getProjectReleaseVersion().equals(saved.getProjectReleaseVersion());
```

Comment:
without app version enforced set, what will be the default behaviour? Will it try to use the manifest file to compare the project release versions? Removing this line looks not

Reply:
if the enforced application version is not set, no upgrade logic will be triggered
In the event of no enforced version set, do we want it to fall back into the old upgrade logic that was relying on the manifest?

Reply:
do we want this:
```

        if (deploymentBuilder.hasEnforcedAppVersion()) {
            return !deploymentBuilder.getEnforcedAppVersion().equals(saved.getVersion());
        }else if (deploymentBuilder.hasProjectManifestSet()){
            return !deployment.getProjectReleaseVersion().equals(saved.getProjectReleaseVersion());
        } else {
           //old code
        }
```

Reply:
yes